openlibrary/templates/account/reading_log.html

```diff
@@ -24,32 +24,39 @@
 $add_metatag(property="og:description", content=og_description)
 $add_metatag(property="og:image", content=meta_photo_url)
 
+<form method="GET" class="olform pagesearchbox">
+  <input type="text" minlength="3" placeholder="$_('Search your reading log')" name="q" value="$(query_param('q', ''))"/>
+  <input type="submit"/>
+</form>
+
 <div class="mybooks-list">
-  <span class="mybooks-tools"><img src="/images/icons/icon_sort.png" alt="$_('Sorting by')" style="margin-right:10px;" width="9" height="11">
-    $if sort_order == 'desc':
-      <strong class="lightgreen">$_("Date Added (newest)")</strong>
-      |
-      <a href="$ctx.path?sort=asc">$_("Date Added (oldest)")</a>
-    $else:
-      <a href="$ctx.path?sort=desc">$_("Date Added (newest)")</a>
-      |
-      <strong class="lightgreen">$_("Date Added (oldest)")</strong>
-  </span>
+  $if q:
+    <span class="search-results-stats">$ungettext('1 hit', '%(count)s hits', item_count, count=commify(item_count))</span>
+  $else:
+    <span class="mybooks-tools"><img src="/images/icons/icon_sort.png" alt="$_('Sorting by')" style="margin-right:10px;" width="9" height="11">
+      $if sort_order == 'desc':
+        <strong class="lightgreen">$_("Date Added (newest)")</strong>
+        |
+        <a href="$changequery(sort='asc')">$_("Date Added (oldest)")</a>
+      $else:
+        <a href="$changequery(sort='desc')">$_("Date Added (newest)")</a>
+        |
+        <strong class="lightgreen">$_("Date Added (oldest)")</strong>
+    </span>
 
-  $:macros.Pager(current_page, total_items, results_per_page=25)
+  $:macros.Pager(current_page, item_count, results_per_page=results_per_page)
   <ul class="list-books">
     $if items:
       $ bookshelf_id = {'want-to-read': 1, 'currently-reading': 2, 'already-read': 3}.get(key, None)
-      $ solr_works = get_solr_works(work['key'] for work in items)
       $ item_number = 1
-      $for item in items:
-        $ work = solr_works.get(item['key']) or item
-        $ decorations = (bookshelf_id and owners_page) and macros.ReadingLogButton(work, read_status=bookshelf_id)
-        $ star_rating = macros.StarRatings(item, redir_url='/account/books/already-read', id=item_number) if include_ratings else None
-        $:macros.SearchResultsWork(work, decorations=decorations, availability=item.get('availability'), rating=star_rating)
+      $# enumerate because the index value of items matches the index of ratings, and this way they can be 'zipped' to including ratings when needed.
+      $for idx, item in enumerate(items):
```

Comment:
I ended up not using `zip` because that results in an empty iterator when no ratings are passed, and ratings are only passed on the already-read page. I tried using a condition to check which page was being shown, but it resulted in mostly the same code twice, with a change to not use `rating=ratings[idx]`. I updated the comment to better reflect why `enumerate` is used, and maybe this is okay?